src/backend/distributed/executor/multi_utility.c

```diff
@@ -3193,6 +3223,77 @@ ExecuteDistributedDDLJob(DDLJob *ddlJob)
 }
 
 
+/*
+ * SetSearchPathToCurrentSearchPathCommand generates a command which can
+ * set the search path to the exact same search path that the issueing node
+ * has.
+ *
+ * If the current search path is null (or doesn't have any valid schemas),
+ * the function returns NULL.
+ */
+static char *
+SetSearchPathToCurrentSearchPathCommand(void)
+{
+	StringInfo setCommand = NULL;
+	char *currentSchemas = CurrentSearchPath();
+
+	if (currentSchemas)
+	{
+		setCommand = makeStringInfo();
+		appendStringInfo(setCommand, "SET search_path TO %s;", currentSchemas);
+		return setCommand->data;
+	}
+
+	return NULL;
+}
+
+
+/*
+ * CurrentSearchPath is a C interface for calling current_schemas(bool) that
+ * PostgreSQL exports.
+ *
+ * CurrentSchemas returns all the schemas in the seach_path that are seperated
+ * with comma (,) sign. The returned string can be used to set the search_path.
+ *
+ * The function omits implicit schemas.
+ *
+ * The function returns NULL if there are no valid schemas in the search_path,
+ * mimicing current_schemas(false) function.
+ */
+static char *
+CurrentSearchPath(void)
+{
+	StringInfo currentSchemas = makeStringInfo();
+	List *searchPathList = fetch_search_path(false);
+	ListCell *searchPathCell;
+	bool schemaAdded = false;
+
+	foreach(searchPathCell, searchPathList)
+	{
+		char *schemaName = NULL;
+
+		if (schemaAdded)
+		{
+			appendStringInfoString(currentSchemas, ",");
+		}
+
+		/* watch out for deleted namespace */
+		schemaName = get_namespace_name(lfirst_oid(searchPathCell));
```

Comment:
But I don't want to add `,` at the end of the string. So, I kept the loop but set `schemaAdded ` to false if `schemaName == NULL`. Does that look good?

(I actually copy & paste'd this code from Postgres code. I cannot find a case where `schemaName` is null. But, keeping it here just in case.)